airflow/api_fastapi/core_api/routes/ui/grid.py

```diff
@@ -0,0 +1,292 @@
+from sqlalchemy.orm import Session
+from typing_extensions import Any
+
+from airflow import DAG
+from airflow.api_fastapi.common.db.common import get_session, paginated_select
+from airflow.api_fastapi.common.parameters import (
+    DateTimeQuery,
+    QueryDagRunRunStatesFilter,
+    QueryDagRunRunTypesFilter,
+    QueryLimit,
+    QueryOffset,
+    SortParam,
+)
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.datamodels.ui.grid import (
+    GridDAGRun,
+    GridDAGRunwithTIs,
+    GridResponse,
+    GridTaskInstance,
+    GridTaskInstanceSummary,
+)
+from airflow.api_fastapi.core_api.openapi.exceptions import create_openapi_http_exception_doc
+from airflow.configuration import conf
+from airflow.exceptions import AirflowConfigException
+from airflow.models import DagRun, TaskInstance
+from airflow.models.baseoperator import BaseOperator
+from airflow.utils.state import TaskInstanceState
+from airflow.utils.task_group import TaskGroup
+
+grid_router = AirflowRouter(prefix="/grid", tags=["Grid"])
+
+
+@grid_router.get(
+    "/",
+    include_in_schema=False,
+    responses=create_openapi_http_exception_doc([status.HTTP_400_BAD_REQUEST, status.HTTP_404_NOT_FOUND]),
+)
+def grid_data(
+    dag_id: str,
+    base_date: DateTimeQuery,
+    run_types: QueryDagRunRunTypesFilter,
+    run_states: QueryDagRunRunStatesFilter,
+    session: Annotated[Session, Depends(get_session)],
+    offset: QueryOffset,
+    request: Request,
+    num_runs: QueryLimit,
+    root: str | None = None,
+    filter_upstream: bool = False,
+    filter_downstream: bool = False,
+) -> GridResponse:
+    """Return grid data."""
+    ## Database calls to retrieve the DAG Runs and Task Instances and validate the data
+    dag: DAG = request.app.state.dag_bag.get_dag(dag_id)
+    if not dag:
+        raise HTTPException(status.HTTP_404_NOT_FOUND, f"Dag with id {dag_id} was not found")
+
+    if root:
+        dag = dag.partial_subset(
+            task_ids_or_regex=root, include_upstream=filter_upstream, include_downstream=filter_downstream
+        )
+
+    if num_runs is None:
+        num_runs = QueryLimit(conf.getint("webserver", "default_dag_run_display_number"))
+
+    # Retrieve, sort and encode the previous DAG Runs
+    base_query = (
+        select(
+            DagRun.run_id,
+            DagRun.queued_at,
+            DagRun.start_date,
+            DagRun.end_date,
+            DagRun.state,
+            DagRun.run_type,
+            DagRun.data_interval_start,
+            DagRun.data_interval_end,
+            DagRun.dag_version_id.label("version_number"),
+        )
+        .select_from(DagRun)
+        .where(DagRun.dag_id == dag.dag_id, DagRun.logical_date <= base_date)
+    )
+
+    dag_runs_select_filter, _ = paginated_select(
+        select=base_query,
+        filters=[
+            run_types,
+            run_states,
+        ],
+        order_by=SortParam(allowed_attrs=["id"], model=DagRun),
```

Comment:
```suggestion
        order_by=SortParam(allowed_attrs=["id"], model=DagRun),
```

SortParam needs an actual value to sort. otherwise it default to the primary key.

Also in the legacy endpoint, these are comming from the `timetable` maybe we should keep it that way.